lib/ohai/plugins/vmware.rb

```diff
@@ -0,0 +1,77 @@
+#
+# Author:: "Dan Robinson" <drobinson@getchef.com>
+# Author:: "Christopher M. Luciano" <cmlucian@us.ibm.com>
+# Copyright:: Copyright (c) 2014 Chef Software, Inc.
+# Copyright (C) 2015 IBM Corp.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Provides a set of attributes for a VMware ESX virtual machine with results
+# obtained from vmware-toolbox-cmd.  VMware Tools must be installed
+# on the virtual machine.
+#
+# Modify the path to vmware-toolbox-cmd in the call to get_vm_attributes for
+# your particular operating system and configuration
+#
+# Example:
+#
+# get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
+#
+
+Ohai.plugin(:Vmware) do
+  provides "vmware"
+
+  def from_cmd(cmd)
+    so = shell_out(cmd)
+    so.stdout.split($/)[0]
+  end
+
+  def get_vm_attributes(vmtools_path)
+    if !File.exist?(vmtools_path)
+      Ohai::Log.debug("#{vmtools_path} not found")
+    else
+      vmware Mash.new
```

Comment:
I bet the pattern you're used to seeing is something like this:

provides 'attribute'

collect_data do
  results = Mash.new
  results['key'] = value
  ...
  attribute results #this sets attribute in the Ohai data mash
end

Ohai doesn't add an attribute mash to its data if there's an = to prevent
collision with variables.

(Sorry, sending from phone.)
On Jun 18, 2015 12:14 PM, "Adam Leff" <notifications@github.com> wrote:

> In lib/ohai/plugins/vmware.rb
> <https://github.com/chef/ohai/pull/551#discussion_r32767367>:
>
> > +# get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
> > +#
> > +
> > +Ohai.plugin(:Vmware) do
> > +  provides "vmware"
> > +
> > +  def from_cmd(cmd)
> > +    so = shell_out(cmd)
> > +    so.stdout.split($/)[0]
> > +  end
> > +
> > +  def get_vm_attributes(vmtools_path)
> > +    if !File.exist?(vmtools_path)
> > +      Ohai::Log.debug("#{vmtools_path} not found")
> > +    else
> > +      vmware Mash.new
>
> Thanks - just seems inconsistent with many of the other plugins,
> especially ones delivered with Chef. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/ohai/pull/551/files#r32767367>.
>